cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -918,34 +922,39 @@ class CompositeTableBuilder {
 // guaranteeing this probability is below 1 in a billion. The fix is 128-bit hashing.
 // See ARROW-17653
 class AsofJoinNode : public ExecNode {
-  // Advances the RHS as far as possible to be up to date for the current LHS timestamp
-  Result<bool> UpdateRhs() {
+  // Advances the RHS as far as possible to be up to date for the current LHS timestamp.
+  // Returns a pair of booleans. The first is if any RHS has advanced. The second is if
+  // all RHS are up to date for LHS.
+  Result<std::pair<bool, bool>> UpdateRhsAndCheckUpToDateWithLhs() {
```

Comment:
Can you return a struct with explanatory member names instead?

Reply:
Addressed.